zebra/zebra_srv6_vty.c

```diff
@@ -259,6 +260,346 @@ DEFUN (show_srv6_locator_detail,
 	return CMD_SUCCESS;
 }
 
+static void do_show_srv6_sid_line(struct ttable *tt, struct zebra_srv6_sid *sid,
+				  bool detail)
+{
+	struct listnode *node;
+	struct zserv *client;
+	char clients[256];
+	char ctx[256] = {};
+	char behavior[256] = {};
+	char alloc_mode_str[10] = {};
+	char locator_name[SRV6_LOCNAME_SIZE];
+	struct vrf *vrf;
+	struct interface *ifp;
+	int ret;
+
+	/* Zclients */
+	if (listcount(sid->client_list)) {
+		bool first = true;
+		int i = 0;
+		for (ALL_LIST_ELEMENTS_RO(sid->client_list, node, client)) {
+			if (first)
+				ret = snprintf(clients + i, sizeof(clients) - i,
+					       "%s(%d)",
+					       zebra_route_string(client->proto),
+					       client->instance);
+			else
+				ret = snprintf(clients + i, sizeof(clients) - i,
+					       ", %s(%d)",
+					       zebra_route_string(client->proto),
+					       client->instance);
+
+			if (ret > 0)
+				i += ret;
+		}
+	}
+
+	/* Behavior */
+	if (sid->locator) {
+		if ((sid->locator->sid_format &&
+		     sid->locator->sid_format->type ==
+			     SRV6_SID_FORMAT_TYPE_USID) ||
+		    (!sid->locator->sid_format &&
+		     CHECK_FLAG(sid->locator->flags, SRV6_LOCATOR_USID))) {
+			switch (sid->ctx->ctx.behavior) {
```

Comment:
Can this be moved to a separate helper function that just returns the string based on the `behavior`?